hibernate-core/src/main/java/org/hibernate/IncludeRemovals.java

```diff
@@ -11,21 +11,32 @@
 import java.util.List;
 
 /**
- * MultiFindOption implementation to specify whether the returned list
- * of entity instances should contain instances that have been
- * {@linkplain Session#remove(Object) marked for removal} in the
- * current session, but not yet deleted in the database.
+ * When {@linkplain SessionChecking} is enabled, this option controls how
+ * to handle entities which are already contained by the persistence context
+ * but which are in a removed state (marked for removal, but not yet flushed).
  * <p>
- * The default is {@link #EXCLUDE}, meaning that instances marked for
- * removal are replaced by null in the returned list of entities when {@link OrderedReturn}
- * is used.
+ * The default is {@link #EXCLUDE}.
  *
- * @see org.hibernate.MultiFindOption
- * @see OrderedReturn
  * @see org.hibernate.Session#findMultiple(Class, List, FindOption...)
  * @see org.hibernate.Session#findMultiple(EntityGraph, List , FindOption...)
+ *
+ * @since 7.2
  */
 public enum IncludeRemovals implements MultiFindOption {
+	/**
+	 * Removed entities are included in the load result.
+	 */
 	INCLUDE,
+	/**
+	 * The default.  Removed entities are excluded from the load result.
+	 * <p/>
+	 * When combined with {@linkplain OrderedReturn#UNORDERED}, the entity is
+	 * simply excluded from the result.
+	 * <p/>
```

Comment:
I keep tripping over this.
**If** by "excluded from the load result" you mean that **no** nulls are present in the load result for removed instances when sessionchecking == enabled, includeremovals == exclude and ordering == unordered, then I can't get this one to work as expected:
e.g. running `MultiLoadTest.testUnorderedMultiLoadFrom2ndLevelCachePendingDeleteReturnRemoved`, with these settings returns a List of size 3, where one element is null, which seems to contradict what follows the IF above.
If what follows the IF above is NOT what you meant then my assumption is wrong.

Reply:
I think EXCLUDE + UNORDERED is really the only thing confusing you?

My memory is just wrong for that case.  It does appear that in this case the null for the removed entity is still included in the result.  That's not what I remember and I have no reasonable explanation why that is the case.  I mean, I can see maybe wanting to avoid needing to resize the list later or something like that, but from a user perspective I think that seems wrong.

Reply:
> My memory is just wrong for that case. It does appear that in this case the null for the removed entity is still included in the result. That's not what I remember and I have no reasonable explanation why that is the case. I mean, I can see maybe wanting to avoid needing to resize the list later or something like that, but from a user perspective I think that seems wrong.
> 
> Given that, I almost think the naming is wrong.

So, change name, reword doc some more, or create a jira if you consider it's a bug?

Reply:
Its not that I consider it a bug per-se.  I just consider it less than obvious behavior.  But I assume its probably always been this way.  Perhaps a third option makes sense, though this third option would depend on the value for `OrderedReturn`.

```java
enum IncludeRemovals {
    /** Legacy false option */
    INCLUDE,
    /** Legacy true option - replace with null */
    REPLACE,
    /** New option. */
    EXCLUDE
}
```

This new option would trigger the more (imo) appropriate behavior -

* With ordered results, we replace with nulls.  This is ofc needed to maintain ordering.
* With unordered results, we'd not have any entry in the result for removed entities.

Reply:
Don't we have a backward compatibility problem with that?

Reply:
With adding a new option?  Why would that be a problem?